traffic_prophet/countmatch/tests/test_neighbour.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+import sklearn.metrics as sklm
+import sklearn.neighbors as skln
+
+from .. import neighbour as nbr
+
+from ...data import SAMPLE_LONLAT
+from ... import cfg
+
+
+class TestNeighbourLocalEuclidean:
+    """Tests Euclidean file-based neighbour finder.  Also tests base class."""
+
+    def setup(self):
+        # There are only 10 items, so this is an all-to-all comparison.
+        self.nle = nbr.NeighbourLocalEuclidean(SAMPLE_LONLAT, 9)
+
+    def test_initialization(self):
+        assert self.nle.data.shape == (10, 3)
+        assert np.array_equal(self.nle.data.columns,
+                              np.array(['ID', 'Lon', 'Lat']))
+        assert np.array_equal(
+            self.nle.data.index.values,
+            self.nle.to_idxs(self.nle.to_ids(self.nle.data.index.values)))
+
+    def test_getxy(self):
+        # Also implicitly tests lonlat_to_xy.
+        lon0, lat0 = cfg.distances['centre_of_toronto']
+
+        lons = np.array([lon0, lon0, lon0 - 1.])
+        lats = np.array([lat0, lat0 + 1., lat0])
+
+        dtr = np.radians(1.)
+        outs = np.array(
+            [[0., 0.],
+             [0., self.nle._r_earth * dtr],
+             [-self.nle._r_earth * dtr * np.cos(np.radians(lat0)), 0.]])
+    
+        assert np.allclose(outs, self.nle.get_xy(lons, lats))
+
+    def test_getneighbours(self):
+        # Check that we're using the correct metric.
+        assert isinstance(self.nle._metric,
+                          skln.dist_metrics.EuclideanDistance)
+
+        # Manually calculate pair-pair distances.
+        xy = self.nle.get_xy(self.nle.data['Lon'].values,
+                             self.nle.data['Lat'].values)
+        distmtx = sklm.pairwise_distances(xy, metric='euclidean')
+
+        # Run neighbour finder.
+        self.nle.get_neighbours()
+
+        # Check that the results are identical (take advantage of the fact that
+        # index is just range(10)).
+        for i in range(distmtx.shape[0]):
+            assert np.array_equal(
+                self.nle.to_idxs(self.nle.data.at[i, 'Neighbours']),
+                np.argsort(distmtx[i])[1:])
+            assert np.allclose(self.nle.data.at[i, 'Distances'],
+                               np.sort(distmtx[i])[1:])
+
+        # Just to show our estimate method is sensible, also check
+        # haversine distances.
```

Comment:
"also check haversine distances" -> "check that Euclidean distances are acceptably close to haversine great circle distances."